internal/incrementalcompiler-core/src/main/scala/sbt/internal/inc/MiniSetupUtil.scala

```diff
@@ -19,13 +20,20 @@ object MiniSetupUtil {
       def sameCompiler = equivComp.equiv(a.compilerVersion, b.compilerVersion)
       def sameOrder = a.order == b.order
       def sameNameHasher = a.nameHashing == b.nameHashing
+      def sameExtra = equivPairs.equiv(a.extra, b.extra)
       sameOutput &&
         sameOptions &&
         sameCompiler &&
         sameOrder && // equivOrder.equiv(a.order, b.order)
-        sameNameHasher
+        sameNameHasher &&
+        sameExtra
     }
   }
+  implicit val equivPairs: Equiv[Array[T2[String, String]]] = new Equiv[Array[T2[String, String]]] {
+    def equiv(a: Array[T2[String, String]], b: Array[T2[String, String]]): Boolean =
+      ((a.toList filter { x => !(x.get1 startsWith "info.") }) sameElements
+        (b.toList filter { x => !(x.get1 startsWith "info.") }))
+  }
```

Comment:
Two `extra`s are considered equivalent if they have the same mapping for all keys that do not start with `info.`?
So the two following `extra`s are equivalent?
```scala
val e1 = Array(("info.foo", "foo"))
val e2 = Array(("info.foo", "bar"), ("info.baz", "hello"))
```

Why?